pkgs/applications/audio/soundconverter/default.nix

```
@@ -0,0 +1,89 @@
+{ lib, fetchurl
+# Optional due to unfree license.
+, faacSupport ? false
+, glib, python3Packages, gtk3, wrapGAppsHook
+, gsettings-desktop-schemas, intltool, xvfb-run
+, gobject-introspection, gst_all_1, fdk-aac-encoder }:
+
+python3Packages.buildPythonApplication rec {
+  pname = "soundconverter";
+  version = "4.0.3";
+
+  src = fetchurl {
+    url = "https://launchpad.net/soundconverter/trunk/${version}/+download/${pname}-${version}.tar.gz";
+    sha256 = "sha256-hzIG/4LD3705erPYvXb7uoRwF9LtKKIKB3jrhpYMsZ0=";
+  };
+
+  buildInputs = [
+    gtk3
+    fdk-aac-encoder
+    gobject-introspection
+    gst_all_1.gst-libav
+    gst_all_1.gst-plugins-base
+    gst_all_1.gst-plugins-good
+    gst_all_1.gst-plugins-ugly
+    (gst_all_1.gst-plugins-bad.override { inherit faacSupport; })
+  ];
+
+  nativeBuildInputs = [
+    intltool
+    wrapGAppsHook
+  ];
+
+  propagatedBuildInputs = [
+    python3Packages.gst-python
+    python3Packages.distutils_extra
+    python3Packages.setuptools
+    python3Packages.pygobject3
+  ];
+
+  checkInputs = [
+    xvfb-run
+  ];
+
+  postPatch = ''
+    substituteInPlace  bin/soundconverter --replace \
+      "DATA_PATH = os.path.join(SOURCE_PATH, 'data')" \
+      "DATA_PATH = '$out/share/soundconverter'"
+  '';
+
+  preCheck = let
+    self = { outPath = "$out"; name = "${pname}-${version}"; };
+    xdgPaths = lib.concatMapStringsSep ":" glib.getSchemaDataDirPath;
+  in ''
+    export HOME=$TMPDIR
+    export XDG_DATA_DIRS=$XDG_DATA_DIRS:${xdgPaths [gtk3 gsettings-desktop-schemas self]}
```

> But wait, how can I use getSchemaDataDirPath with $out?

I don't think you can.

```suggestion
  preCheck = let
    xdgPaths = lib.concatMapStringsSep ":" glib.getSchemaDataDirPath;
  in ''
    export HOME=$TMPDIR
    export XDG_DATA_DIRS=$XDG_DATA_DIRS:${xdgPaths [gtk3 gsettings-desktop-schemas self]}:"$out/share/gsettings-schemas/${pname}-${version}"
```
like this. I am pretty sure that the self way will not work with overlays which overwrite pname+version correct.

---

Why would it not work? How is it different than what you're doing? The value of `XDG_DATA_DIRS` comes out the same.

I don't see a difference other than making the line longer and harder to read.

---

If you override the package rec is not updated recursively and the old things would still be in place.

---

Oh, I see. If that's the behavior of `rec` then indeed the path would not be effected and that would be an issue.

But that's not how it works:
```
 > nix repl .
Welcome to Nix 2.6.1. Type :? for help.

Loading '.'...
Added 15702 variables.

nix-repl> :b pkgs.soundconverter.overrideAttrs (_: { pname = "whatever"; version = "6.6.6"; })

This derivation produced the following outputs:
  out -> /nix/store/hwhs0fdi7h1iiglh3n9lql8p5r8110xp-soundconverter-4.0.3
```
Does `overrideAttrs `even work with `python3Packages.buildPythonApplication`?

---

You want to use [overridePythonAttrs](https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#overriding-python-packages-overriding-python-packages) but I think this is fine either way the more I think about it.

---

Oh, you're right, `overridePythonAttrs` does work, and indeed it does fail then due to a schema error:
```
> (test.py:307): GLib-GIO-ERROR **: 16:45:35.619: Settings schema 'org.soundconverter' is not installed
```

---

Not sure if we want to do something about that or not.

---

Well, I really don't see WHY someone would want to change the `pname` of this derivation.

I see no reason why that would be useful in any way.

---

If you have an overlay which overwrites version this will fall apart and will be hard to debug but there are more issues regarding that.

---

Thanks for merging, but for future reference, what's the correct way to pass the version to `src` if not with `rec`?